packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFInstanceManager.m

```diff
@@ -3,70 +3,143 @@
 // found in the LICENSE file.
 
 #import "FWFInstanceManager.h"
+#import <objc/runtime.h>
+
+// Attaches to an object to receive a callback when the object is deallocated.
+@interface FWFFinalizer : NSObject
+@property(nonatomic) long identifier;
+// Callbacks are no longer made once FWFInstanceManager is inaccessible.
+@property(nonatomic, weak) FWFOnDeallocCallback callback;
++ (void)attachToInstance:(NSObject *)instance
+          withIdentifier:(long)identifier
+                callback:(FWFOnDeallocCallback)callback;
++ (void)detachFromInstance:(NSObject *)instance;
+@end
+
+@implementation FWFFinalizer
+- (instancetype)initWithIdentifier:(long)identifier callback:(FWFOnDeallocCallback)callback {
+  self = [self init];
+  if (self) {
+    _identifier = identifier;
+    _callback = callback;
+  }
+  return self;
+}
+
++ (void)attachToInstance:(NSObject *)instance
+          withIdentifier:(long)identifier
+                callback:(FWFOnDeallocCallback)callback {
+  FWFFinalizer *finalizer = [[FWFFinalizer alloc] initWithIdentifier:identifier callback:callback];
+  objc_setAssociatedObject(instance, _cmd, finalizer, OBJC_ASSOCIATION_RETAIN);
+}
+
++ (void)detachFromInstance:(NSObject *)instance {
+  objc_setAssociatedObject(instance, @selector(attachToInstance:withIdentifier:callback:), nil,
+                           OBJC_ASSOCIATION_ASSIGN);
+}
+
+- (void)dealloc {
+  self.callback(self.identifier);
```

Comment:
You should not use `self` in dealloc.

Given the usage pattern of this class, I would strongly recommend you go against our usual policy for new ObjC code and not make properties at all, and instead just declare them as ivars. There's no case where the properties should be called.